src/Confluent.Kafka/Timestamp.cs

```diff
@@ -85,12 +87,13 @@ public override int GetHashCode()
         /// </param>
         /// <returns>
         ///     The milliseconds unix timestamp corresponding to <paramref name="dateTime"/>
+        ///     rounded to the previous millisecond if dateTime precision is below millisecond
         /// </returns>
         public static long DateTimeToUnixTimestampMs(DateTime dateTime)
         {
             checked
             {
-                return (long)(dateTime.ToUniversalTime() - UnixEpochUtcDateTime).TotalMilliseconds;
+                return dateTime.ToUniversalTime().Ticks / TimeSpan.TicksPerMillisecond - UnixEpochMilliseconds;
```

Comment:
now you have me wondering if we should be rounding to the nearest millisecond or rounding down. I'm currently favouring rounding down (as I had it before, and as you continue going forward) because I think this type of binning is easier to think about and I prefer it if times never get adjusted into the future. Somewhat arbitrary as I don't expect this level of precision is really ever required in practice.